commit 822115787ae7f19ca9352fcb1568dbb34133e67d
Author: Crineu Tres <crineu.tres@ufsc.br>
Date:   Mon May 30 23:24:20 2022 -0300

    Mudança lugar -> empreendimento

diff --git a/src/main/java/com/checkapp/controle/InspecaoControle.java b/src/main/java/com/checkapp/controle/InspecaoControle.java
index 27fa053..7603227 100644
--- a/src/main/java/com/checkapp/controle/InspecaoControle.java
+++ b/src/main/java/com/checkapp/controle/InspecaoControle.java
@@ -48,11 +48,11 @@ public class InspecaoControle implements Serializable {
     private int aba;
     
     private Item item;
-    private Empreendimento lugar;
+    private Empreendimento empreendimento;
     private Categoria categoria;
     
     private List<Categoria> categorias;
-    private List<SelectItem> comboLugar;
+    private List<SelectItem> comboEmpreendimentos;
     private List<Categoria> listaDeCategoria;
     
     @Autowired
@@ -73,11 +73,11 @@ public class InspecaoControle implements Serializable {
     //para pesquisar no banco antes de carregar a tela- como um construtor de uma classe de Entidade, mas tem em todas as classes
     @PostConstruct
     public void iniciar() {
-            lugar = null;
-            modelInspecoes = null;
-            aba = 0;
+        aba = 0;
+        empreendimento = new Empreendimento();
+        modelInspecoes = null;
         inspecao = new Inspecao();
-        carregarComboBoxLugar();
+        carregarComboBoxEmpreendimentos();
         listaDeCategoria = categoriaRepositorio.pesquisarCategoriaPorItem();
 
 //        avaliacoes = new ArrayList<>();
@@ -109,11 +109,11 @@ public class InspecaoControle implements Serializable {
         Optional<Inspecao> inspecoes = inspecaoRepositorio.findById(inspecao.getId());
     }
     
-    private void carregarComboBoxLugar() {
+    private void carregarComboBoxEmpreendimentos() {
         List<Empreendimento> lugares = lugarRepositorio.findAll();
-        comboLugar = new ArrayList<>();
+        comboEmpreendimentos = new ArrayList<>();
         for (Empreendimento lug : lugares) {
-            comboLugar.add(new SelectItem(lug.getId(), lug.getNome()));
+            comboEmpreendimentos.add(new SelectItem(lug.getId(), lug.getNome()));
         }
     }
     
@@ -154,7 +154,7 @@ public class InspecaoControle implements Serializable {
     public void salvar() {
         try {
             inspecao.setNome("Teste");
-            inspecao.setEmpreendimento(lugar);
+            inspecao.setEmpreendimento(empreendimento);
             inspecao.setDataEhora(GregorianCalendar.getInstance().getTime());
             
             inspecaoRepositorio.save(inspecao);
@@ -210,8 +210,8 @@ public class InspecaoControle implements Serializable {
 //    }
     public void onTabChange(TabChangeEvent event) {
         if (event.getTab().getTitle().equals("Novo")) {
-            if (comboLugar == null) {
-                carregarComboBoxLugar();
+            if (comboEmpreendimentos == null) {
+                carregarComboBoxEmpreendimentos();
             }
             if (listaDeCategoria == null) {
                 carregarListaItem();
@@ -262,12 +262,12 @@ public class InspecaoControle implements Serializable {
         this.listaDeCategoria = listaDeCategoria;
     }
     
-    public List<SelectItem> getComboLugar() {
-        return comboLugar;
+    public List<SelectItem> getComboEmpreendimentos() {
+        return comboEmpreendimentos;
     }
     
-    public void setComboLugar(List<SelectItem> comboLugar) {
-        this.comboLugar = comboLugar;
+    public void setComboEmpreendimentos(List<SelectItem> comboEmpreendimentos) {
+        this.comboEmpreendimentos = comboEmpreendimentos;
     }
     
     public Inspecao getInspecao() {
@@ -302,15 +302,12 @@ public class InspecaoControle implements Serializable {
         this.inspecaoRepositorio = inspecaoRepositorio;
     }
     
-    public Empreendimento getLugar() {
-        if (lugar == null) {
-            lugar = new Empreendimento();
-        }
-        return lugar;
+    public Empreendimento getEmpreendimento() {
+        return empreendimento;
     }
     
-    public void setLugar(Empreendimento lugar) {
-        this.lugar = lugar;
+    public void setEmpreencimento(Empreendimento empreendimento) {
+        this.empreendimento = empreendimento;
     }
     
     public int getAba() {
diff --git a/src/main/webapp/novo.xhtml b/src/main/webapp/novo.xhtml
index 598583e..100bb2a 100644
--- a/src/main/webapp/novo.xhtml
+++ b/src/main/webapp/novo.xhtml
@@ -20,11 +20,11 @@
                                      contentStyleClass="ui-fluid">
   
                             <p:outputLabel for="lugar" value="Local:" />
-                            <p:selectOneMenu id="lugar" value="#{inspecaoC.lugar.id}" 
+                            <p:selectOneMenu id="lugar" value="#{inspecaoC.empreendimento.id}" 
                                              required="true" requiredMessage="Nome empreendimento é obrigatório!"
                                              filter="true" filterMatchMode="startsWith">
                                 <f:selectItem itemLabel="Selecione o empreendimento:"/>
-                                <f:selectItems value="#{inspecaoC.comboLugar}"/>
+                                <f:selectItems value="#{inspecaoC.comboEmpreendimentos}"/>
                             </p:selectOneMenu>
                             
 <!--                            <p:accordionPanel value="#{inspecaoC.listaDeCategoria}" var="categoria">
diff --git a/src/main/webapp/novo_3.xhtml b/src/main/webapp/novo_3.xhtml
index 3f63627..a50e871 100644
--- a/src/main/webapp/novo_3.xhtml
+++ b/src/main/webapp/novo_3.xhtml
@@ -66,7 +66,7 @@
                                              required="false" requiredMessage="Local é obrigatório!"
                                              filter="true" filterMatchMode="startsWith">
                                 <f:selectItem itemLabel="Selecione o local:"/>
-                                <f:selectItems value="#{inspecaoC.comboLugar}"/>
+                                <f:selectItems value="#{inspecaoC.comboEmpreendimentos}"/>
                             </p:selectOneMenu>
 
                             <p:outputLabel for="item" value="Item:" />
diff --git a/src/main/webapp/novo_4.xhtml b/src/main/webapp/novo_4.xhtml
index 7afec0e..1841c18 100644
--- a/src/main/webapp/novo_4.xhtml
+++ b/src/main/webapp/novo_4.xhtml
@@ -65,7 +65,7 @@
                                              required="true" requiredMessage="Nome empreendimento é obrigatório!"
                                              filter="true" filterMatchMode="startsWith">
                                 <f:selectItem itemLabel="Selecione o local:"/>
-                                <f:selectItems value="#{inspecaoC.comboLugar}"/>
+                                <f:selectItems value="#{inspecaoC.comboEmpreendimentos}"/>
                             </p:selectOneMenu>
                            
                             <!--primeira coluna-->
